components/layout_2020/flow/mod.rs

```diff
@@ -1152,10 +1152,10 @@ impl NonReplacedFormattingContext {
             self.style.clone(),
             layout.fragments,
             content_rect.into(),
-            pbm.padding.into(),
-            pbm.border.into(),
-            margin,
-            clearance,
+            pbm.padding,
+            pbm.border,
+            margin.into(),
+            clearance.map(|t| t.into()),
```

Comment:
You can make make `solve_clearance_and_inline_margins_avoiding_floats` return an `Option<Au>` for the clearance instead of converting back and forth.